### PR TITLE
PyPI memary integration 

### PR DESCRIPTION
**Pull request by kingjulio8238**
Requirement for PyPi Integration - setup.py

### CODE REVIEW

**Comment by kingjulio8238** [comment]
See PR #30 

Additional edits to do before packaging:

1. Update the folder structure to have it as memary/memary folder 
2. Add 'tests' folder but leave empty for now

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "llama-index-graph-stores-neo4j==0.1.1",
+    "llama-index-legacy==0.9.48",
+    "llama-index-lms-openai==0.1.5",
+    "llama-index-multi-modal-llms-openai==0.1.3",
+    "llama-index-program-openai==0.1.3",
+    "llama-index-question-gen-openai==0.1.2",
+    "llama-index-readers-file==0.1.4",
+    "langchain==0.1.12",
+    "langchain-openai==0.0.8",
+    "llama-index-lms-perplexity==0.1.3",
```
Comment (seyeong-han) [inline pyproject.toml:40]: "llama-index-llms-perplexity==0.1.3",

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "pyvis==0.3.2",
+    "streamlit==1.31.1",
+    "llama-index==0.10.11", 
+    "llama-index-agent-openai==0.1.5",
+    "llama-index-core==0.1.12",
+    "llama-index-embeddings-openai==0.1.5",
+    "llama-index-graph-stores-nebula==0.1.2",
+    "llama-index-graph-stores-neo4j==0.1.1",
+    "llama-index-legacy==0.9.48",
+    "llama-index-lms-openai==0.1.5",
```
Comment (seyeong-han) [inline pyproject.toml:33]: "llama-index-llms-openai==0.1.5",

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "Operating System :: OS Independent",
+]
+dependencies = [
+    "neo4j==5.17.0",
+    "python-dotenv==1.0.1",
+    "pyvis==0.3.2",
+    "streamlit==1.31.1",
+    "llama-index==0.10.11", 
+    "llama-index-agent-openai==0.1.5",
+    "llama-index-core==0.1.12",
```
Comment (seyeong-han) [inline pyproject.toml:28]: "llama-index-core==0.10.12",

File: dist/memary_test-0.1.4-py3-none-any.whl
Comment (seyeong-han) [inline dist/memary_test-0.1.4-py3-none-any.whl:1]: You should not upload this wheel file to test `pip install`.
You can test it right after building this file locally within a new environment, `pip install dist/memary_test-0.1.4-py3-none-any.whl`
Please run and test `streamlit run app.py` before push the codes.
And you don't have to push this dist directory.

**Review by seyeong-han — CHANGES_REQUESTED** [review]
I hope you guys test the codes first and upload them.
Thanks.

**Comment by kevinl424** [comment]
After running `python3 -m build` from the root directory and starting a new virtual environment, running `pip install dist/memary_test-0.1.3-py3-none-any.whl', and changing the `ChatAgent` import to `from memary.agent.chat_agent import ChatAgent` (all the directory stuff would no longer be needed), the streamlit app works.

Following review
- update the two images mentioned in above comment
- push package

**Comment by kingjulio8238** [comment]
Note: `pip install dist/memary-0.1.3-py3-none-any.whl` works instead of memary_test

**Review by seyeong-han — COMMENTED** [review]
I should have told you earlier, but I think it's better to rename `src` as `memary`.
There is no problem with operating codes, but the structure looks redundant.
It depends on your preference. I'm fine with this structure also.

**Comment by kevinl424** [comment]
> I should have told you earlier, but I think it's better to rename `src` as `memary`. There is no problem with operating codes, but the structure looks redundant. It depends on your preference. I'm fine with this structure also.

Was thinking under `src` may be better long term just in case we add some other module that goes along with memary, although not sure if that will happen. Prefer to keep it as it is but definitely open to changing it too.

**Review by seyeong-han — DISMISSED** [review]
Thanks @kevinl424 @kingjulio8238 .
I tested PyPI build files and all good.
Also, I fixed the README image links to use the local `diagrams` images.
Now, it is good to publish PyPI package after merging this branch.

**Review by kevinl424 — APPROVED** [review]

**Review by kevinl424 — APPROVED** [review]

**Review by seyeong-han — APPROVED** [review]
LGTM
